### PR TITLE
WIP Remove deprecated newrelic_plugins_alert_condition resource

### PR DESCRIPTION
**Pull request by jaymcgrath**
As explained in #1327, New Relic Plugins are reaching EOL June 16th, 2021. 

This seeks to replace the already deprecated newrelic_plugins_alert_condition resource with an error message, as suggested in Step 4 of the [Hashicorp best practices guide for removing resources](https://www.terraform.io/docs/extend/best-practices/deprecations.html#provider-data-source-or-resource-removal)

I've also removed references to this resource from the docs index and replaced the resource's docs page with a message about the removal.

I'm marking this as a WIP PR as I'm having some issues getting my environment back up to speed locally, and hoping that, given how it's all code/test removal and docs updates,  it just builds in the CI 🤞

### CODE REVIEW

**Comment by jthurman42** [comment]
This is an interesting case as TF wise we should merge this into a 3.x release, but the API is going to stop working regardless...  @kidk do we have a plan for `newrelic_dashboards` as well here? Good to lump them together.

**Comment by ctrombley** [comment]
We have a deprecation of the plugins resource coming through in the next couple days.  Closer to the actual EOL we'll move forward with a more definitive change like this one.  Thanks @jaymcgrath!

cc  @mbazhlekova @kidk

File: website/docs/d/plugin.html.markdown [hunk 1]
```diff
@@ -21,27 +21,6 @@ data "newrelic_plugin" "foo" {
   guid = "com.example.my-plugin"
 }
```
Comment (ctrombley) [inline website/docs/d/plugin.html.markdown:23]: Add an EOL message to this data source similar to the one we added for the resource?

**Comment by ctrombley** [comment]
Closing in favor of #1350.

**Pull request closed without merging**